lib/src/iterable_extensions.dart

```diff
@@ -166,6 +166,17 @@ extension IterableExtension<T> on Iterable<T> {
     }
   }
 
+  /// Returns an iterable containing only the non-`null` results
+  /// of applying the given [transform] function to each element in this iterable.
+  Iterable<R> mapNotNull<R extends Object>(R? Function(T) transform) sync* {
```

Comment:
This is basically a combination of `map` and `whereNotNull`. Or a variant of `expand` where there is always zero or one element in the resulting list.

I've often wanted a variant of `map` with a built-in `where` to skip some elements, and this function would satisfy that.

So, definitely useful.